core/src/main/java/com/google/instrumentation/trace/TraceConfig.java

```diff
@@ -0,0 +1,212 @@
+     *
+     * @return the global default max number of {@code Link} entries per {@code Span}.
+     */
+    public abstract int getMaxNumberOfLinks();
+
+    private static Builder builder() {
+      return new AutoValue_TraceConfig_TraceParams.Builder();
+    }
+
+    /**
+     * Returns a {@link Builder} initialized to the same property values as the current instance.
+     *
+     * @return a {@link Builder} initialized to the same property values as the current instance.
+     */
+    public abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    public abstract static class Builder {
+
+      /**
+       * Sets the global default {@code Sampler}. It must be not {@code null} otherwise {@link
+       * #build()} will throw an exception.
+       *
+       * @param sampler the global default {@code Sampler}.
+       * @return this.
+       */
+      public abstract Builder setSampler(Sampler sampler);
+
+      /**
+       * Sets the global default max number of attributes per {@link Span}.
+       *
+       * @param maxNumberOfAttributes the global default max number of attributes per {@link Span}.
+       *     It must be positive otherwise {@link #build()} will throw an exception.
+       * @return this.
+       */
+      public abstract Builder setMaxNumberOfAttributes(int maxNumberOfAttributes);
+
+      /**
+       * Sets the global default max number of {@link Annotation} events per {@link Span}.
+       *
+       * @param maxNumberOfAnnotations the global default max number of {@link Annotation} events
+       *     per {@link Span}. It must be positive otherwise {@link #build()} will throw an
+       *     exception.
+       * @return this.
+       */
+      public abstract Builder setMaxNumberOfAnnotations(int maxNumberOfAnnotations);
+
+      /**
+       * Sets the global default max number of {@link NetworkEvent} events per {@link Span}.
+       *
+       * @param maxNumberOfNetworkEvents the global default max number of {@link NetworkEvent}
+       *     events per {@link Span}. It must be positive otherwise {@link #build()} will throw an
+       *     exception.
+       * @return this.
+       */
+      public abstract Builder setMaxNumberOfNetworkEvents(int maxNumberOfNetworkEvents);
+
+      /**
+       * Sets the global default max number of {@link Link} entries per {@link Span}.
+       *
+       * @param maxNumberOfLinks the global default max number of {@link Link} entries per {@link
+       *     Span}. It must be positive otherwise {@link #build()} will throw an exception.
+       * @return this.
+       */
+      public abstract Builder setMaxNumberOfLinks(int maxNumberOfLinks);
+
+      public abstract TraceParams autoBuild();
+
+      /**
+       * Builds and returns a {@code TraceParams} with the desired values.
+       *
+       * @return a {@code TraceParams} with the desired values.
+       * @throws NullPointerException if the sampler is null.
```

Comment:
IllegalStateException does not seem like the correct exception "Signals that a method has been invoked at an illegal or inappropriate time." I would replace both of these with:
@throws IllegalArgumentException if sampler is {@code null} or any of the max numbers are not positive.